.pre-commit-config.yaml

```diff
@@ -0,0 +1,39 @@
+# .pre-commit-config.yaml
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.1.0
+    hooks:
+    # Identify invalid files
+    - id: check-ast
+    - id: check-yaml
+    - id: check-json
+    # git checks
+    - id: check-merge-conflict
+    - id: check-added-large-files
+    - id: detect-private-key
+    - id: check-case-conflict
+    # Python checks
+    - id: check-docstring-first
+    - id: debug-statements
+    - id: requirements-txt-fixer
+    - id: name-tests-test
+    # General quality checks
+    - id: mixed-line-ending
+    - id: trailing-whitespace
+      args: [--markdown-linebreak-ext=md]
+      exclude: 'rasgotransforms/rasgotransforms/transforms'
+    - id: check-executables-have-shebangs
+    - id: end-of-file-fixer
+      exclude: 'rasgotransforms/rasgotransforms/transforms'
+    - id: no-commit-to-branch
+  - repo: https://github.com/psf/black
+    rev: 22.3.0
+    hooks:
+      - id: black
+        args: [--line-length=120, --skip-string-normalization]
```

Comment:
Any reason to use these and not the defaults? Never used black in detail before.

Are there other options we may want to change?

Reply:
Line length devs can fight over all day long, going through our code bases, it seemed that most things were soft wrapped at 120 characters so set it there (Both PyCharm and VSCode have ruler marks at 120, so it's a common setting.) 

`--skip-string-normalization` we will hopefully remove when we have time (just means all strings will go to `"` instead of `'` and usually want to do some manual cleanup after it.)